client/src/graphics/draw.rs

```diff
@@ -253,9 +271,11 @@ impl Draw {
     ///
     /// Submits commands that wait on `image_acquired` before writing to `framebuffer`'s color
     /// attachment.
+    #[allow(clippy::too_many_arguments)] // Every argument is of a different type, making this less of a problem.
     pub unsafe fn draw(
         &mut self,
         mut sim: Option<&mut Sim>,
+        gui_state: &GuiState,
```

Comment:
Take a `yakui::paint::PaintDom` here, and store the `GuiState` at a higher level. That way the Vulkan rendering code can remain fully isolated from the abstract GUI logic, which will maintain better separation of concerns and probably make incremental compiles happier besides.

Reply:
I hadn't thought about the incremental compiles aspect, but I'm always for anything that keeps Hypermine compilation speeds fast.